src/app-layout/visual-refresh-toolbar/drawer/global-drawer.tsx

```diff
@@ -0,0 +1,137 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useEffect, useRef } from 'react';
+import clsx from 'clsx';
+
+import { InternalButton } from '../../../button/internal';
+import PanelResizeHandle from '../../../internal/components/panel-resize-handle';
+import { fireNonCancelableEvent, NonCancelableEventHandler } from '../../../internal/events';
+import customCssProps from '../../../internal/generated/custom-css-properties';
+import { usePrevious } from '../../../internal/hooks/use-previous';
+import { createWidgetizedComponent } from '../../../internal/widgets';
+import { getLimitedValue } from '../../../split-panel/utils/size-utils';
+import { AppLayoutProps } from '../../interfaces';
+import { AppLayoutInternals } from '../interfaces';
+import { useResize } from './use-resize';
+
+import sharedStyles from '../../resize/styles.css.js';
+import testutilStyles from '../../test-classes/styles.css.js';
+import styles from './styles.css.js';
+
+interface AppLayoutGlobalDrawerImplementationProps {
+  appLayoutInternals: AppLayoutInternals;
+  show: boolean;
+  activeGlobalDrawer:
+    | (AppLayoutProps.Drawer & { onShow?: NonCancelableEventHandler; onHide?: NonCancelableEventHandler })
+    | undefined;
+}
+
+export function AppLayoutGlobalDrawerImplementation({
+  appLayoutInternals,
+  show,
+  activeGlobalDrawer,
+}: AppLayoutGlobalDrawerImplementationProps) {
+  const {
+    ariaLabels,
+    globalDrawersFocusControl,
+    isMobile,
+    placement,
+    onActiveGlobalDrawersChange,
+    onActiveDrawerResize,
+    minGlobalDrawersSizes,
+    maxGlobalDrawersSizes,
+    activeGlobalDrawersSizes,
+    drawersOpenQueue,
+  } = appLayoutInternals;
+  const drawerRef = useRef<HTMLDivElement>(null);
+  const activeDrawerId = activeGlobalDrawer?.id ?? '';
+  const prevShow = usePrevious(show);
+
+  useEffect(() => {
+    if (activeGlobalDrawer?.preserveInactiveContent && !!prevShow !== show) {
+      if (show) {
+        fireNonCancelableEvent(activeGlobalDrawer?.onShow);
+      } else {
+        fireNonCancelableEvent(activeGlobalDrawer?.onHide);
+      }
+    }
+  }, [activeGlobalDrawer, prevShow, show]);
+
+  const computedAriaLabels = {
+    closeButton: activeGlobalDrawer ? activeGlobalDrawer.ariaLabels?.closeButton : ariaLabels?.toolsClose,
+    content: activeGlobalDrawer ? activeGlobalDrawer.ariaLabels?.drawerName : ariaLabels?.tools,
+  };
+
+  const activeDrawerSize = activeGlobalDrawersSizes[activeDrawerId ?? ''] ?? 0;
```

Comment:
This `?? ''` part seems concerning. We might mix up state between multiple drawers. This would be safer

```suggestion
  const activeDrawerSize = (activeDrawerId ? activeGlobalDrawersSizes[activeDrawerId] : 0) ?? 0;
```